lib/next_rails.rb

```diff
@@ -1,8 +1,24 @@
+# frozen_string_literal: true
+
 require "next_rails/gem_info"
 require "next_rails/version"
 require "next_rails/bundle_report"
 require "deprecation_tracker"
 
 module NextRails
-  # Your code goes here...
+
+  # This method will check your environment
+  # (e.g. `ENV['BUNDLE_GEMFILE]`) to determine whether your application is
+  # running with the next set of dependencies or the current set of dependencies.
+  #
+  # @return [Boolean]
+  def self.next?
+    @@next_bundle_gemfile ||= ENV["BUNDLE_GEMFILE"] == "Gemfile.next"
```

Comment:
the only thing I noticed is this can't memoize the `false` value

maybe something like this?

```ruby
module NextRails
  @@next_bundle_gemfile = nil

  # This method will check your environment
  # (e.g. `ENV['BUNDLE_GEMFILE]`) to determine whether your application is
  # running with the next set of dependencies or the current set of dependencies.
  #
  # @return [Boolean]
  def self.next?
    return @@next_bundle_gemfile unless @@next_bundle_gemfile.nil?
    
    @@next_bundle_gemfile = ENV["BUNDLE_GEMFILE"] == "Gemfile.next"
  end

  # This method will reset the @@next_bundle_gemfile variable. Then next time
  # you call `NextRails.next?` it will check the environment once again.
  def self.reset_next_bundle_gemfile
    @@next_bundle_gemfile = nil
  end
end
```

I feel checking `unless ...nil?` is faster than the string comparison?

Reply:
we may use `defined?` for false and nil values


```suggestion
    return @@next_bundle_gemfile if defined? @@next_bundle_gemfile
    
    @@next_bundle_gemfile ||= ENV["BUNDLE_GEMFILE"] == "Gemfile.next"
```

Reply:
I played a little bit around it and opened a [pr](https://github.com/fastruby/next_rails/pull/102/files)